exec/cnex/nstring.c

```diff
@@ -145,6 +179,17 @@ void string_resizeString(TString *s, size_t n)
     s->length = n;
 }
 
+TString *string_reverseString(TString *s, size_t start, size_t end)
+{
+    TString *r = string_createString(end-start);
+    for (size_t i = start, e = end; e > start; e--, i++) {
```

Comment:
I'm not convinced this loop isn't off by one. When `e` starts, it points to `end`, which is the last character to reverse, correct? But then the continuation test is `e > start`, so `e` will never actually equal `start` inside the loop. So the `start` character never gets copied.

Reply:
There was absolutely an issue with this loop.  It has now been fixed.